src/ui/win32/DialogBase.hh

```diff
@@ -12,48 +12,54 @@ namespace win32 {
 
 class DialogBase
 {
-public:    
+public:
+    DialogBase() = delete;
+    DialogBase(const DialogBase&) noexcept = delete;
+    DialogBase& operator=(const DialogBase&) noexcept = delete;
+    DialogBase(DialogBase&&) noexcept = delete;
+    DialogBase& operator=(DialogBase&&) noexcept = delete;
+
     /// <summary>
     /// Creates the dialog window (but does not show it).
     /// </summary>
     /// <param name="sResourceId">The resource identifier defining the dialog.</param>
     /// <param name="pDialogClosed">Callback to call when the dialog is closed.</param>
     /// <returns>Handle of the window.</returns>
-    HWND CreateDialogWindow(LPTSTR sResourceId, IDialogPresenter* pDialogPresenter);
+    /*_NODISCARD */HWND CreateDialogWindow(_In_ const LPCTSTR sResourceId,
```

Comment:
Either include `_NODISCARD` or keep it. Leaving it with a comment is confusing - is it supposed to be there or not?